File: frontend/lib/models/event.dart
import 'user.dart';
class Event {

  final int id;
  final String name;
  final String description;
  final String location;
  final double ticketPrice;
  final int capacity;
  final int available;
  final String eventDate;
  final String imageUrl;
  final String category;
  final bool active;
  final int organizerId;
  final User? organizer;
  

  Event({
    required this.id,
    required this.name,
    required this.description,
    required this.location,
    required this.ticketPrice,
    required this.capacity,
    required this.available,
    required this.eventDate,
    required this.imageUrl,
    required this.category,
    required this.active,
    required this.organizerId,
    this.organizer, 
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      location: json['location'] ?? '',
      ticketPrice: (json['ticketPrice'] ?? 0).toDouble(),
      capacity: json['capacity'] ?? 0,
      available: json['available'] ?? 0,
      eventDate: json['eventDate'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
      category: json['category'] ?? '',
      active: json['active'] ?? true,
      organizerId: json['organizer']?['id'] ?? 0,
      organizer: json['organizer'] != null ? User.fromJson(json['organizer']) : null, 
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'location': location,
      'ticketPrice': ticketPrice,
      'capacity': capacity,
      'eventDate': eventDate,
      'imageUrl': imageUrl,
      'category': category,
    };
  }

  Event copyWith({
    int? id,
    String? name,
    String? description,
    String? location,
    double? ticketPrice,
    int? capacity,
    int? available,
    String? eventDate,
    String? imageUrl,
    String? category,
    bool? active,
    int? organizerId,
    User? organizer,
  }) {
    return Event(
      id: id ?? this.id,
      name: name ?? this.name,
      description: description ?? this.description,
      location: location ?? this.location,
      ticketPrice: ticketPrice ?? this.ticketPrice,
      capacity: capacity ?? this.capacity,
      available: available ?? this.available,
      eventDate: eventDate ?? this.eventDate,
      imageUrl: imageUrl ?? this.imageUrl,
      category: category ?? this.category,
      active: active ?? this.active,
      organizerId: organizerId ?? this.organizerId,
      organizer: organizer ?? this.organizer,
    );
  }
}

File: backend/src/main/java/com/str/backend/model/Event.java
package com.str.backend.model;

import java.util.List;
import java.util.Set;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToMany;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;

@Entity
@Table(name = "events")
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;
    private String description;
    private String location;
    private Double ticketPrice;
    private Integer capacity;
    private Integer available;
    private String eventDate;
    private String imageUrl;
    private String category;
    private Boolean active = true;

    @ManyToOne
    @JoinColumn(name = "organizer_id")
    private User organizer;

    @ManyToMany(mappedBy = "events")
    private Set<User> users;

    @OneToMany(mappedBy = "event")
    @JsonIgnore
    private List<Ticket> tickets;

    public Event() {}

    public Event(String name, String description, String location, Double ticketPrice, 
                 Integer capacity, String eventDate, String category) {
        this.name = name;
        this.description = description;
        this.location = location;
        this.ticketPrice = ticketPrice;
        this.capacity = capacity;
        this.available = capacity;
        this.eventDate = eventDate;
        this.category = category;
        this.active = true;
    }

    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getLocation() { return location; }
    public void setLocation(String location) { this.location = location; }

    public Double getTicketPrice() { return ticketPrice; }
    public void setTicketPrice(Double ticketPrice) { this.ticketPrice = ticketPrice; }

    public Integer getCapacity() { return capacity; }
    public void setCapacity(Integer capacity) { this.capacity = capacity; }

    public Integer getAvailable() { return available; }
    public void setAvailable(Integer available) { this.available = available; }

    public String getEventDate() { return eventDate; }
    public void setEventDate(String eventDate) { this.eventDate = eventDate; }

    public String getImageUrl() { return imageUrl; }
    public void setImageUrl(String imageUrl) { this.imageUrl = imageUrl; }

    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }

    public Boolean getActive() { return active; }
    public void setActive(Boolean active) { this.active = active; }

    public User getOrganizer() { return organizer; }
    public void setOrganizer(User organizer) { this.organizer = organizer; }

    public Set<User> getUsers() { return users; }
    public void setUsers(Set<User> users) { this.users = users; }

    public List<Ticket> getTickets() { return tickets; }
    public void setTickets(List<Ticket> tickets) { this.tickets = tickets; }
}